homeassistant/components/nasweb/helper.py

```diff
@@ -0,0 +1,46 @@
+"""Helper functions for NASweb integration."""
+
+from collections.abc import Mapping
+import logging
+from typing import Any
+
+from homeassistant.components.http import ApiConfig
+from homeassistant.core import HomeAssistant
+
+from .const import CONF_HA_ADDRESS, PUSH_API_ENDPOINT
+from .coordinator import NotificationCoordinator
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def initialize_notification_coordinator(
+    hass: HomeAssistant,
+) -> NotificationCoordinator | None:
+    """Initialize and set up NotificationCoordinator instance."""
+    notify_coordinator = NotificationCoordinator()
+    try:
+        _LOGGER.debug("Adding push notification handler")
+        hass.http.app.router.add_post(
```

Comment:
We don't allow integrations do directly interact with hass.http.router to register endpoints.
You should either register a view, or maybe more appropriate, a webhook to allow the API to interact with Home Assistant.